### ISSUE TITLE
The all parameter in API roles should not mix guest and authenticated users w/o role

### ISSUE THREAD

**Issue by rezib** [edf-hpc/slurm-web#103]
Currently, here is the spec for all parameter in REST API configuration file:

```
[roles]
# The `all` role can have 2 values:
# - none, restrict access to authenticated users only
# - all, free access to non-sensible data
all = all
```

With `all = none`, both guest users and authenticated users w/o other role are not allowed to access the cluster.
With `all = all`, both guest users and authenticated users w/o other role are allowed to access the cluster.

Some admins may want to allow authenticated users w/o other role but not guests.

Then, the possible values should be something like:

* `none`: both guests and authenticated users w/o other role are denied
* `guest`: guests are allowed
* `authenticated`: authenticated users w/o other role are allowed
* `guest,authenticated`: both guests and authenticated users w/o other role are allowed

These guest, authenticated values should be usable for other roles as well. An admin should be able to allow guest to be admins as well.

Maybe `all` parameter name should be also renamed to `basic` for more clarity.

**Comment by rezib** [comment]
Default value should be false.

**Comment by rezib** [comment]
This is a duplicate of #104, which has been fixed with 181973e in 2.1.
